analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,485 @@
+
+# don't include the normal cells that were provided as reference in scaling 
+reference_cells_only <- all_cnv_df |> 
+  dplyr::filter(stringr::str_detect(subcluster, "reference")) |> 
+  dplyr::mutate(has_cnv = NA,
+                z_scaled_cnv = NA)
+
+unknown_cells_only <- all_cnv_df |>
+  dplyr::filter(stringr::str_detect(subcluster, "unknown")) |> 
+  # get total number of chromosomes with cnvs per cell and then scale that number 
+  dplyr::mutate(has_cnv = rowSums(across(has_cnv_cols)),
+                z_scaled_cnv = scale(has_cnv))
+
+# combine back together for plotting 
+cnv_df <- bind_rows(reference_cells_only, unknown_cells_only)
+```
+
+
+```{r}
+# plot the total number of CNVs
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = has_cnv)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  scale_color_viridis_c()
+```
+This shows that the total CNVs are highest in the group of cells that correspond to the same cells that are manually annotated as tumor cells, which is promising. 
+Those cells in grey (bottom left) are cells used as a reference. 
+If we were to use total CNVs we would need some sort of cut off for the number. 
+Let's look at the distribution of total CNVs and also the scaled total CNVs to see if we can identify a cut off. 
+
+```{r}
+ggplot(cnv_df, aes(x = has_cnv)) +
+  geom_density() +
+  theme_bw() +
+  labs(x = "Total number of CNVs")
+
+ggplot(cnv_df, aes(x= z_scaled_cnv)) + 
+  geom_density() +
+  theme_bw() +
+  labs(x = "z-scaled chromosome CNV count")
+```
+
+Looking at the scaled number of CNVs it does look like we could use 0 as the cutoff and call any cell that has > 0 CNVs (after scaling) as tumor and all other cells as normal. 
+
+In the below plot we label all non-reference cells as either tumor or normal. 
+Cells that are NA are those that were already classified as reference cells. 
+
+```{r}
+# add cnv classification 
+cnv_df <- cnv_df |> 
+  dplyr::mutate(cnv_classification = dplyr::if_else(z_scaled_cnv > 0, "Tumor", "Normal"))
+
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = cnv_classification)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() 
+```
+
+Another option is to use the scaled proportion of CNVs which tells us the proportion of genes on that chromosome that are part of the CNV.  
+
+```{r}
+proportion_cols <- names(cnv_df)[startsWith(colnames(cnv_df), "proportion_scaled_cnv")]
+
+# get total number of genes per chromosome to use in weighted mean 
+chr_weights <- final_cnv_obj@gene_order |> 
+  as.data.frame() |> 
+  dplyr::count(chr) |> 
+  # only keep chr 1-22, drops MT and GL chr
+  dplyr::filter(chr %in% glue::glue("chr{seq(1,22)}")) |> 
+  dplyr::pull(n)
+
+# get mean proportion of cnvs per cell weighted by chromsome size (number of genes)
+cnv_df <- cnv_df |> 
+  dplyr::rowwise() |> 
+  # calculate weighted mean, excluding normal cells 
+  dplyr::mutate(mean_proportion = ifelse(!is.na(has_cnv), weighted.mean(across(proportion_cols), num_genes_df$n), NA)) |> 
```

Comment:
You did it in one step this time, but I would break up across lines for clarity

Reply:
You are using `num_genes_df` here, but I think this should be `chr_weights` after the code? I also think this is probably not quite right, as the MT genes have not been excluded from the proportion columns.

Rather than filtering out those chromosomes above, I might just set their weights to 0 so you don't have to worry about mismatched lengths?

Reply:
Sorry the `num_genes_df` is leftover from when that's what it was named. Also I have removed the MT genes from all columns so the dimensions should match up correctly.